### ISSUE TITLE
Rule idea: favor const

### ISSUE THREAD

**Issue by hgl** [eslint/eslint#2333]
Doesn't allow you to do this:
```js
function foo() {
  let bar = 1;
  alert(bar); // bar is never rebound, should use const to define bar
}
```

Should be beneficial when you want to encourage the use of `const`.

**Comment by brigand** [comment]
+1, and along similar lines to disallow var entirely (with let/const it isn't needed).

**Comment by nzakas** [comment]
Yup, this was already discussed in #1617. 

@brigand there's already a `no-var` rule

**Comment by brigand** [comment]
Oh wow, how did I miss that? Thanks nzakas.

**Comment by hgl** [comment]
I wonder if a reversed version also makes sense: no const, only let is allowed.

**Comment by nzakas** [comment]
That would be a bit strange. Would you really want to disallow const?

**Comment by hgl** [comment]
If const becomes a trend somehow (afaik, iojs already forces it), and your code base only uses let, I guess you might want to keep it "pure" by disallowing const.

**Comment by nzakas** [comment]
I don't see that as a general case. You'd probably need a custom rule for that.

**Comment by mysticatea** [comment]
I will try to make this.
I like const!
